### PR TITLE
matrix-synapse-plugins.matrix-synapse-mjolnir-antispam: init at 0.1.17

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change
Add the [mjolnir antispam](https://github.com/matrix-org/mjolnir#synapse-module) plugin for synapse. I recently migrated my matrix homeserver to NixOS and am running mjolnir as well (created a derivation for it, but that one needs to be cleaned up a little - I've modified the upstream source to be able to load additional configuration files, e.g. the secret parts like the password of the superuser). If there is interest to add mjolnir to nixpkgs i can create a PR for it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123879` at 8d7b4507 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>matrix-synapse-plugins.matrix-synapse-mjolnir-antispam</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix:18:3:
```
   |
18 |   doCheck = false; # no tests
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix:18:3:
```
   |
18 |   doCheck = false; # no tests
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-uBI5AllXWgl3eL60WZ/j11Tt7QpY7CKcmFQOU74/Qjs=";
+  };
+
+  sourceRoot = "./source/synapse_antispam";
+
+  propagatedBuildInputs = [ matrix-synapse ];
+
+  doCheck = false; # no tests
+  pythonImportsCheck = [ "mjolnir" ];
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix:19]: Missing meta.

Comment (jojosch) [reply pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix:19]: Seems like none of the other plugins have a `meta` attribute - but i can add one here.

File: pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  meta = with lib; {
+    description = "AntiSpam / Banlist plugin to be used with mjolnir";
+    longDescription = ''
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix:23]: Can we add a homepage or link to github repo?

Comment (jojosch) [reply pkgs/servers/matrix-synapse/plugins/mjolnir-antispam.nix:23]: added a link to the mjolnir readme

**Comment by sumnerevans** [comment]
Do we also need to package the non-synapse module version of this? (Looking at this example: https://github.com/matrix-org/mjolnir#build-it-alternative-installation)

**Pull request merged**